vscode-dotnet-runtime-library/src/Acquisition/GenericDistroSDKProvider.ts

```diff
@@ -0,0 +1,183 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+import { IDistroDotnetSDKProvider } from './IDistroDotnetSDKProvider';
+import { DotnetDistroSupportStatus } from './LinuxVersionResolver';
+import { VersionResolver } from './VersionResolver';
+import * as path from 'path';
+
+export class GenericDistroSDKProvider extends IDistroDotnetSDKProvider {
+
+    public async installDotnet(fullySpecifiedVersion : string): Promise<string>
+    {
+        const supportStatus = await this.getDotnetVersionSupportStatus(fullySpecifiedVersion);
+        if(supportStatus === DotnetDistroSupportStatus.Microsoft)
+        {
+            const preinstallCommands = this.myVersionPackages()[this.preinstallCommandKey];
+            const preparationResult = (await this.commandRunner.execute(preinstallCommands))[0];
+        }
+
+        let command = this.myDistroCommands()[this.installCommandKey];
+        const sdkPackage = this.myDotnetVersionPackages(fullySpecifiedVersion)[this.sdkKey];
+        command = command.replace('{0}', sdkPackage);
+        const commandResult = await this.commandRunner.execute(command);
+
+        return commandResult[0];
+    }
+
+    public async getInstalledGlobalDotnetPathIfExists() : Promise<string | null>
+    {
+        const commandResult = await this.commandRunner.execute(this.myDistroCommands()[this.currentInstallPathCommandKey]);
+        return commandResult[0];
+    }
+
+    public async dotnetPackageExistsOnSystem(fullySpecifiedDotnetVersion : string) : Promise<boolean>
+    {
+        let command = this.myDistroCommands()[this.packageLookupCommandKey];
+        const sdkPackage = this.myDotnetVersionPackages(this.JsonDotnetVersion(fullySpecifiedDotnetVersion))[this.sdkKey];
+        command = command.replace('{0}', sdkPackage);
+        const commandResult = await this.commandRunner.execute(command);
+
+        const noPackageResult = 'no packages found';
+        return commandResult[0].includes(noPackageResult);
```

Comment:
It'd be good to make sure this is case-insensitive

Reply:
Also, does this work in other locales?

Reply:
Created an issue to investigate later. https://github.com/dotnet/vscode-dotnet-runtime/issues/1453